src/mds/objects.py

```diff
@@ -406,48 +406,62 @@ async def delete_object(
         403: if fence/indexd returns a 403 unauthorized response
         500: if fence/indexd does not return 204 or 403 or there is an error deleting metadata
     """
+    # Attempt to get the row, then attempt delete the row from Metadata table
+    metadata_obj = (
+        await Metadata.select("guid", "data").where(Metadata.guid == guid).gino.first()
+    )
+    if metadata_obj:
+        metadata_obj = dict(metadata_obj)
+    await (
+        Metadata.delete.where(Metadata.guid == guid).returning(*Metadata).gino.first()
```

Comment:
i think `.returning(*Metadata)` already returns the object, so you wouldn't have to query it first, but i haven't checked

Reply:
You are right. Let me change it. Thanks for the tip.